api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -593,6 +588,23 @@ public void mergePatients(Patient preferred, Patient notPreferred) throws APIExc
 		Context.getPersonService().savePersonMergeLog(personMergeLog);
 	}
 	
+	private void requireNoActiveOrderOfSameType(Patient patient1, Patient patient2) {
+		String messageKey = "Patient.merge.cannotHaveSameTypeActiveOrders";
+		List<Order> ordersByPatient1 = Context.getOrderService().getAllOrdersByPatient(patient1);
+		List<Order> ordersByPatient2 = Context.getOrderService().getAllOrdersByPatient(patient2);
+		ordersByPatient1.forEach((Order order1) -> {
+			ordersByPatient2.forEach((Order order2) -> {
+				if (order1.isActive() && order2.isActive() && order1.getOrderType().equals(order2.getOrderType())) {
```

Comment:
Do you mean to say that:
Active 'Drug Order' of 'Same chemical' (for clarity) should not exist in both patients while merging?